interfaces/builtin/desktop.go

```diff
@@ -240,7 +240,8 @@ func (iface *desktopInterface) AppArmorConnectedPlug(spec *apparmor.Specificatio
 	// Allow mounting document portal
 	var buf bytes.Buffer
 	fmt.Fprintf(&buf, "  # Mount the document portal\n")
-	fmt.Fprintf(&buf, "  mount options=(bind) /run/user/[0-9]*/doc/by-app/snap.%s/ -> /run/user/[0-9]*/doc/,\n", plug.Snap().Name())
+	// TODO parallel-install: use of proper instance/store name
+	fmt.Fprintf(&buf, "  mount options=(bind) /run/user/[0-9]*/doc/by-app/snap.%s/ -> /run/user/[0-9]*/doc/,\n", plug.Snap().InstanceName())
```

Comment:
CC @jhenstridge can you confirm that XDG document portal allows underscore in application names?

For some context. We are introducing snap instances where you can create an instance of a snap "foo" as "foo_extra" where "extra" is the instance key. Such instances can possibly track different revisions but surely get separate state and per-snap and per-snap-user data. This PR introduces a simple mass rename from the snap name (e.g. "foo") to snap instance name "foo_extra"

Reply:
Looking at the `xdp_is_valid_app_id` function, it looks like underscores are acceptable:

https://github.com/flatpak/xdg-desktop-portal/blob/master/src/xdp-utils.c#L845

The other place where we deal with snap-specific application IDs is parsing the package name out of the AppArmor security label:

https://github.com/flatpak/xdg-desktop-portal/blob/master/src/xdp-utils.c#L441

That is splitting the label on periods, so shouldn't care about underscores.